Source/controls/plrctrls.cpp

```diff
@@ -1249,16 +1250,104 @@ struct RightStickAccumulator {
+	if (event.type == SDL_MOUSEWHEEL)
+		return event.wheel.which == SDL_TOUCH_MOUSEID ? ControlTypes::VirtualGamepad : ControlTypes::KeyboardAndMouse;
+	if (IsAnyOf(event.type, SDL_FINGERDOWN, SDL_FINGERUP, SDL_FINGERMOTION))
+		return ControlTypes::VirtualGamepad;
+	if (event.type == SDL_CONTROLLERAXISMOTION && (leftStickX != 0 || leftStickY != 0 || rightStickX != 0 || rightStickY != 0))
+		return ControlTypes::Gamepad;
+	if (event.type >= SDL_CONTROLLERBUTTONDOWN && event.type <= SDL_CONTROLLERDEVICEREMAPPED)
+		return ControlTypes::Gamepad;
+	if (IsAnyOf(event.type, SDL_JOYDEVICEADDED, SDL_JOYDEVICEREMOVED))
+		return ControlTypes::Gamepad;
+#endif
+	if (event.type == SDL_JOYAXISMOTION && (leftStickX != 0 || leftStickY != 0 || rightStickX != 0 || rightStickY != 0))
+		return ControlTypes::Gamepad;
+	if (event.type >= SDL_JOYBALLMOTION && event.type <= SDL_JOYBUTTONUP)
+		return ControlTypes::Gamepad;
 
-bool IsAutomapActive()
+	return ControlTypes::None;
+}
+
+float rightStickLastMove = 0;
+
+bool ContinueSimulatingMouse(const SDL_Event &event, const ControllerButtonEvent &gamepadEvent)
 {
-	return AutomapActive && leveltype != DTYPE_TOWN;
+	if (IsAutomapActive())
+		return false;
+
+#if !defined(USE_SDL1) && !defined(JOY_AXIS_RIGHTX) && !defined(JOY_AXIS_RIGHTY)
+	if (IsAnyOf(event.type, SDL_JOYAXISMOTION, SDL_JOYHATMOTION, SDL_JOYBUTTONDOWN, SDL_JOYBUTTONUP)) {
+		if (!GameController::All().empty())
+			return true;
+	}
+#endif
+
+	if (rightStickX != 0 || rightStickY != 0 || rightStickLastMove != 0) {
+		rightStickLastMove = rightStickX + rightStickY; // Allow stick to come to a reset with out breaking simulation
+		return true;
+	}
+
+	if (IsAnyOf(gamepadEvent.button, ControllerButton_BUTTON_RIGHTSTICK, ControllerButton_BUTTON_BACK)) {
+		return true;
+	}
+
+	return false;
 }
 
-bool IsMovingMouseCursorWithController()
+} // namespace
+
+void DetectInputMethod(const SDL_Event &event, const ControllerButtonEvent &gamepadEvent)
 {
-	return rightStickX != 0 || rightStickY != 0;
+	ControlTypes inputType = GetInputTypeFromEvent(event);
+
+#ifdef __vita__
+	if (inputType == ControlTypes::VirtualGamepad) {
+		inputType = ControlTypes::Gamepad;
+	}
+#endif
+
+#if HAS_KBCTRL == 1
+	if (inputType == ControlTypes::KeyboardAndMouse && IsNoneOf(gamepadEvent.button, ControllerButton_NONE, ControllerButton_IGNORE)) {
+		inputType = ControlTypes::Gamepad;
+	}
+#endif
+
+	if (ControlMode == ControlTypes::KeyboardAndMouse && inputType == ControlTypes::Gamepad && ContinueSimulatingMouse(event, gamepadEvent)) {
+		return;
+	}
+
+	if (inputType != ControlTypes::None && inputType != ControlMode) {
+		ControlMode = inputType;
+
+#ifndef USE_SDL1
+		if (ControlMode != ControlTypes::KeyboardAndMouse) {
+			if (IsHardwareCursor())
+				SetHardwareCursor(CursorInfo::UnknownCursor());
+		} else {
+			ResetCursor();
+		}
+#endif
+
+		CalculatePanelAreas();
+	}
+} // namespace

```

Comment:
```suggestion
}
```

Reply:
Done :)

Reply:
[llvm-namespace-comment](https://clang.llvm.org/extra/clang-tidy/checks/llvm-namespace-comment.html) is enabled in the clang-tidy config at the moment, this gets flagged (as a warning) by VS when it runs the clang-tidy checks

Reply:
This was the end of a function where the namespace curly brace had ended up being reused.